early-results-test/src/main/java/com/hazelcast/jet/tests/earlyresults/EarlyResultsTest.java

```diff
@@ -100,20 +101,31 @@ protected void teardown(Throwable t) throws Exception {
     static class VerificationContext {
 
         private final int windowSize;
-        private Map<String, TickerWindow> tickerMap = new HashMap<>();
+        private final ILogger logger;
+        private final Map<String, TickerWindow> tickerMap;
 
-        VerificationContext(int windowSize) {
+        VerificationContext(ILogger logger, int windowSize) {
+            this.logger = logger;
             this.windowSize = windowSize;
+            this.tickerMap = new HashMap<>();
         }
 
         void verify(KeyedWindowResult<String, Long> result) {
             TickerWindow tickerWindow = tickerMap.computeIfAbsent(result.getKey(), TickerWindow::new);
+            // we have a result after the window advanced
+            // ignore if it is an early result, fail otherwise
+            if (result.start() < tickerWindow.start) {
+                assertTrue(result.isEarly());
```

Comment:
would be good to have some message here: "a final result older than previous final result" (if this is correct). However in this specific test it's not possible I think.

Reply:
added one